packages/main/src/CalendarHeader.js

```diff
@@ -154,6 +177,51 @@ class CalendarHeader extends UI5Element {
 		}
 	}
 
+	_getDisplayedSecondaryMonthText(month) {
+		const localeData = getCachedLocaleDataInstance(getLocale());
+		const pattern = localeData.getIntervalPattern();
+		const secondaryMonthsNames = getCachedLocaleDataInstance(getLocale()).getMonthsStandAlone("abbreviated", this.secondaryCalendarType);
+		const secondaryMonthsNamesWide = getCachedLocaleDataInstance(getLocale()).getMonthsStandAlone("wide", this.secondaryCalendarType);
+
+		if (month.startMonth === month.endMonth) {
+			return {
+				text: localeData.getMonths("abbreviated", this.secondaryCalendarType)[month.startMonth],
+				textInfo: localeData.getMonths("wide", this.secondaryCalendarType)[month.startMonth],
+			};
+		}
+
+		return {
+			text: pattern.replace(/\{0\}/, secondaryMonthsNames[month.startMonth]).replace(/\{1\}/, secondaryMonthsNames[month.endMonth]),
+			textInfo: pattern.replace(/\{0\}/, secondaryMonthsNamesWide[month.startMonth]).replace(/\{1\}/, secondaryMonthsNamesWide[month.endMonth]),
+		};
+	}
+
+	_getDisplayedSecondaryMonths(localDate) {
+		let firstDate = CalendarDate.fromLocalJSDate(localDate);
+		firstDate.setDate(1);
+		firstDate = new CalendarDate(firstDate, this.secondaryCalendarType);
+
+		const startMonth = firstDate.getMonth();
+		let lastDate = CalendarDate.fromLocalJSDate(localDate, this.primaryCalendarType);
+
+		lastDate.setDate(this._daysInMonth(lastDate));
+		lastDate = new CalendarDate(lastDate, this.secondaryCalendarType);
+		const endMonth = lastDate.getMonth();
+		return { startMonth, endMonth };
+	}
+
+	_daysInMonth(calendarDate) {
```

Comment:
Actually I found this function in the CalendarUtils module in the openui5 project so it clearly doesn't belong to the CalendarHeader component here as well.